modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/queue/KafkaEncoder.scala

```diff
@@ -0,0 +1,26 @@
+package ch.epfl.bluebrain.nexus.iam.service.queue
+
+import ch.epfl.bluebrain.nexus.iam.core.acls._
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.auto._
+import io.circe.syntax._
+import io.circe.{Encoder, Json, Printer}
+
+// $COVERAGE-OFF$
+trait KafkaEncoder {
```

Comment:
We should add a custom Identity `Encoder` with the `@id` which is the combination with `origin` and `group` to be a qualified id

Reply:
@umbreak do you mean something like this?
```
protected implicit val userRefEncoder: Encoder[UserRef] = deriveEncoder[UserRef].mapJsonObject { obj: JsonObject =>
    (for {
      origin <- obj("origin")
      subject <- obj("subject")
    } yield s"$origin/$subject") match {
      case Some(id) => obj.add("@id", Json.fromString(id))
      case None => obj
    }
  }
```

Reply:
I thinking of using the following representations:
_Anonymous_:
```
{
  "@context": "{{base}}/context",
  "@type": "Anonymous"
}
```
_AuthenticatedRef_:
```
{
  "@context": "{{base}}/context",
  "@type": "AuthenticatedRef",
  "realm": "{{realm}}" // optional
}
```
_UserRef_:
```
{
  "@context": "{{base}}/context",
  "@type": "UserRef",
  "@id": "{{base}}/realms/{{realm_name}}/users/{{sub}}",
  "realm": "{{realm}}",
  "sub": "{{sub}}"
}
```
_GroupRef_:
```
{
  "@context": "{{base}}/context",
  "@type": "GroupRef",
  "@id": "{{base}}/realms/{{realm_name}}/groups/{{group}}",
  "realm": "{{realm}}",
  "group": "{{group}}"
}
```
Thoughts?

Reply:
That make sense.
In this case should `{{base}}` be e.g. https://bbp-nexus.epfl.ch/dev/v0/acls/ 
and what should the `{{realm_name}}` in this case? should we try to parse it from the `{{realm}}`?

Reply:
Yes, the base value is correct.

Short answer: from config.
Longer answer:
I think each provider configuration should define a `realm_name` property such that the system can work with multiple providers at the same time. A call to `/oauth2/authorize` could take a query param like `realm=bbp` which selects the downstream provider for authentication.